pkg/cmd/pulumi/convert/convert.go

```diff
@@ -457,13 +482,153 @@ func runConvert(
 	return nil
 }
 
-func newPluginContext(cwd string) (*plugin.Context, error) {
-	sink := diag.DefaultSink(os.Stderr, os.Stderr, diag.FormatOptions{
-		Color: cmdutil.GetGlobalColorization(),
-	})
-	pluginCtx, err := plugin.NewContext(sink, sink, nil, nil, cwd, nil, true, nil)
+// getPackagesToGenerateSdks parses the pcl files back in to read the package
+// blocks and for sdk generation.
+func getPackagesToGenerateSdks(
+	sourceDirectory string,
+) (map[string]*schema.PackageDescriptor, hcl.Diagnostics, error) {
+	// TODO use the package block to get the packages and parameterize them and generate SDKs.
+	files, err := os.ReadDir(sourceDirectory)
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not read sourceDirectory: %w", err)
+	}
+
+	parser := hclsyntax.NewParser()
+	_, err = pcl.ParseFiles(parser, sourceDirectory, files)
```

Comment:
Maybe make this `ParseDirectory`? Would fit nicely into `BindDirectory` and save you duplicating the `ReadDir` code here and in the `pcl` package.